docs/design/deviceshifu/design-deviceShifu-lwm2m.md

```diff
@@ -0,0 +1,157 @@
+# DeviceShifu lwM2M Design
+
+deviceShifu lwM2M allow shifu connect device using [lwM2M protocol].
+This document outlines the design for integrating the [Lightweight Machine to Machine (lwM2M) protocol](https://omaspecworks.org/what-is-oma-specworks/iot/lightweight-m2m-lwm2m/) into the DeviceShifu framework to allow for the management of IoT devices.
+
+## Goals
+
+### Design Goal
+
+- Create a deviceShifu-lwm2m type allow user to connect device using lwM2M protocol.
+- lwM2M protocol support both `read` and `write` requests.
+- Resource observation or Notification
+- lwM2M protocol using v1.0.x version.
+- lwM2M protocol under UDP.
+- 
+### Design Non-Goal
+
+- Support lwM2M v1.1.x or later version.
+- Datagram Transport Layer Security (DTLS) support.
+- Over TCP or other protocol.
+- Bootstrap Server.
+
+## Detail Design
+
+The `deviceShifu-lwM2M` will represent each lwM2M object as an instruction that can be accessed through a RESTful API. The API will use the POST method for write operations and the GET method for read operations. The supported data formats for lwM2M v1.0 are TLV, JSON, Plain text, and Opaque.
+
+deviceShifu-lwM2M's lwM2M server will handle Register, Update, De-register, Read, Write, Observe and Notify.
```

Comment:
you mean external LwM2M server? Like mosquitto broker for MQTT. Not part of deviceShifu

Reply:
No it is a part of deviceShifu, for devices are LwM2M client and deviceShifu is server which device can connect to us